app/Http/Controllers/ConversionMetricController.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Wiki;
+use Carbon\CarbonImmutable;
+use Illuminate\Http\Response;
+use Illuminate\Http\Request;
+
+class ConversionMetricController extends Controller
+{
+
+    private string $fileName="conversion_metric_for_all_wikis.csv";
+
+    /**
+     * Produce a downloadable csv file with conversion metrics for all wikis.
+     */
+    public function index(Request $request)
+    {
+        $allWikis = Wiki::all();
+        $current_date = CarbonImmutable::now();
+        $output = [];
+        
+
+        foreach ($allWikis as $wiki) {
+            $lifecycleEvents = $wiki->wikiLifecycleEvents()->first();
+            $wikiLastEditedTime = null;
+            $wikiFirstEditedTime = null;
+            if ($lifecycleEvents !== [] && $lifecycleEvents->count() > 0) {
```

Comment:
I'm probably being easily confused but I think this also doesn't work.

should be just `!empty(..)` I think ?

Reply:
assuming the [docs](https://www.php.net/manual/en/function.empty.php) aren't outdated it would be the same I think (also same as `false != empty($x), since `empty()` is supposed to return a boolean; I was just playing it safe

Reply:
see the test failure below and also me trying to clear up my confusion at https://3v4l.org/5iG1l